hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/TestLinuxContainerExecutorWithMocks.java

```diff
@@ -311,6 +312,12 @@ public void testStartLocalizer() throws IOException {
           .build());
 
       List<String> result=readMockParams();
+
+      if (Shell.isJavaVersionAtLeast(17)) {
+        assertTrue(result.remove("--add-exports=java.base/sun.net.dns=ALL-UNNAMED"));
+        assertTrue(result.remove("--add-exports=java.base/sun.net.util=ALL-UNNAMED"));
```

Comment:
MAPREDUCE-7456 defines 

```
  private static final String ADDITIONAL_JDK17_PLUS_OPTIONS =
       "--add-opens=java.base/java.lang=ALL-UNNAMED " +
       "--add-exports=java.base/sun.net.dns=ALL-UNNAMED " +
       "--add-exports=java.base/sun.net.util=ALL-UNNAMED";
```

Why should only the last two be removed?

BTW, could you please leave a few comments that at least mention MAPREDUCE-7456 ?

Reply:
Thank you @pan3793 .

JDK17 options are defined at several locations.
Other than the one you mentioned, they are also defined in org.apache.hadoop.yarn.server.nodemanager.containermanager.localizer.ContainerLocalizer which only adds 
these two, and apparently these tests trigger that one.

If I added all three, the tests would fail, as --add-opens=java.base/java.lang=ALL-UNNAMED is never added for those test cases.

This also bothers me, I have opened HADOOP-19505 earlier to track this. (Though I have no immediate plans to work on that)

Reply:
As fot comments, most tests in TestContainerLocalizer already have similar ifs for JDK17 (I guess these were just added later, and never tested with JDK17). 
Would you like me to add comments to those as well ?

Reply:
> Would you like me to add comments to those as well ?

Yes, please do that, I think the brief comments should help future explores easy to understand the test code for this case

Reply:
Done.